components/I18nAddressFields.js

```diff
@@ -0,0 +1,198 @@
+ * How Etsy Localizes addresses https://codeascraft.com/2018/09/26/how-etsy-localizes-addresses/
+ * Form i18n techniques https://medium.com/flexport-design/form-internationalization-techniques-3e4d394cd7e5 */
+
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import AddressFormatter from '@shopify/address';
+import { Field, useFormikContext } from 'formik';
+import { get, isEmpty, orderBy, pick, truncate } from 'lodash';
+import memoizeOne from 'memoize-one';
+
+import LoadingPlaceholder from './LoadingPlaceholder';
+import StyledInput from './StyledInput';
+import StyledInputField from './StyledInputField';
+import StyledSelect from './StyledSelect';
+
+/** Constants */
+
+/** Countries present in InputTypeCountry dropdown but not Shopify's API.
+ * All except Antarctica (AQ) are U.S. territories and use that address format.
+ * The Shopify default address format is also U.S. format therefore for all
+ * of these we use the U.S. default.
+ * All language codes in locales.js are supported by the Shopify API 👍
+ */
+const missingCountries = ['AS', 'AQ', 'GU', 'MH', 'FM', 'MP', 'PW', 'PR', 'VI'];
+const addressFormatter = new AddressFormatter('EN');
+
+const wrangleAddressData = addressInfo => {
+  const formLayout = addressInfo.formatting.edit;
+  const necessaryFields = ['address1', 'address2', 'city', 'zip', 'province'];
+
+  // Get form fields in correct order for the chosen country
+  const matches = formLayout.match(/([A-Za-z])\w+/g).filter(match => necessaryFields.includes(match));
+
+  // Change field names to match https://github.com/Shopify/quilt/blob/master/packages/address/src/utilities.ts
+  const mappedMatches = matches.map(match => {
+    if (match === 'zip') {
+      return 'postalCode';
+    } else if (match === 'province') {
+      return 'zone';
+    } else {
+      return match;
+    }
+  });
+
+  const addressFormFields = Object.entries(addressInfo.labels)
+    .filter(entry => mappedMatches.includes(entry[0]))
+    .sort((a, b) => {
+      return mappedMatches.indexOf(a[0]) - mappedMatches.indexOf(b[0]);
+    });
+
+  // Check if we need to render drop-down list of "zones" (i.e. provinces, states, etc.)
+  const zones = get(addressInfo, 'zones', []);
+  if (mappedMatches.includes('zone') && !isEmpty(zones)) {
+    const zoneIndex = addressFormFields.find(idx => idx[0] === 'zone');
+    zoneIndex.push(addressInfo.zones);
+  }
+
+  return addressFormFields;
+};
+
+/** Checks if the key from labels is also present in optionalLabels */
+const isFieldOptional = (addressInfo, fieldName) => {
+  return !Object.keys(addressInfo.optionalLabels).includes(fieldName);
+};
+
+/** Component to be used in forms that require addresses that need some validation
+ * i.e. Expenses and Contributions. Must be used with Formik. */
+const I18nAddressFields = ({ selectedCountry }) => {
+  const formik = useFormikContext();
```

Comment:
Not using Formik here would make things more flexible, and being able to use the component in non-formik forms would be a nice added value. What about using standard `onChange`/`value` props? Like `StyledSelect`, we would still be able to use it in a formik form like this:

```jsx
<Field name="location">
  {({ field }) => (
    <I18nAddressFields
      value={field.value}
      onChange={({ value, error }) => {
        if (error) {
          field.setError(error);
        } else {
          field.setValue(value);
        }
      }}
  )}
</Field>
```